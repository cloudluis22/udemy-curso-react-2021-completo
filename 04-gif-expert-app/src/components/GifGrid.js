import React from "react";
import useFetchGifs from "../hooks/useFetchGifs";
import GifGridItem from './GifGridItem'
import PropTypes from 'prop-types';

const GifGrid = ({category: cat, delete: del}) => {

  const { loading, data: images } = useFetchGifs(cat);

  return (

    <div>

        <div className="bar">
          <h3>{cat}</h3>
        <span id="close" onClick = {() => del(cat)}>&times;</span>
        </div>

      {loading && <p>Cargando</p>}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>

    </div>

  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
  delete: PropTypes.func.isRequired
}

export default GifGrid;