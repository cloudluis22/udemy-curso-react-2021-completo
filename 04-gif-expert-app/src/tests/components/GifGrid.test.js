import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el componente de <GifGrid />.', () => {

    const category = 'Doom Eternal';
    const del = jest.fn();

    beforeEach(() => {
        del.mockClear();
    })

    test('Verificar que el componente se renderice correctamente.', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} delete={del}/>)
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar los items cuando se cargan los gifs utilizando useFetchGifs', () => {

        const gifs = [{
            id: 'ID Ejemplo',
            URL: 'https://exampleurl.com',
            title: 'Titulo Ejemplo'
        }]

        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} delete={del} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);

    });

});