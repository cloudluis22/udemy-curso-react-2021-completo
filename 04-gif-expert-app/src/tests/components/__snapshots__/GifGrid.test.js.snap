// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de <GifGrid />. Debe mostrar los items cuando se cargan los gifs utilizando useFetchGifs 1`] = `
<div>
  <div
    className="bar"
  >
    <h3>
      Doom Eternal
    </h3>
    <span
      id="close"
      onClick={[Function]}
    >
      ×
    </span>
  </div>
  <div
    className="card-grid"
  />
</div>
`;

exports[`Pruebas en el componente de <GifGrid />. Verificar que el componente se renderice correctamente. 1`] = `
<div>
  <div
    className="bar"
  >
    <h3>
      Doom Eternal
    </h3>
    <span
      id="close"
      onClick={[Function]}
    >
      ×
    </span>
  </div>
  <p>
    Cargando
  </p>
  <div
    className="card-grid"
  />
</div>
`;
